tests/admin_scripts/tests.py

```diff
@@ -2116,19 +2116,26 @@ def test_invalid_name(self):
 
     def test_invalid_target_name(self):
         """startapp validates that target dir name is a valid Python identifier."""
-        os.mkdir('app_dir')
-        self.addCleanup(shutil.rmtree, 'app_dir')
-
-        for bad_target in ('app_dir/invalid.name', 'app_dir/7invalid_name'):
+        for bad_target in ('invalid.dir_name', '7invalid_dir_name'):
             args = ['startapp', 'valid_name', bad_target]
+
             os.mkdir(bad_target)
 
             out, err = self.run_django_admin(args)
+            shutil.rmtree(bad_target)
```

Comment:
This line and `os.mkdir(bad_target)` are unnecessary, we don't need to create directories for this test.